### PR TITLE
Jakarta Servlet 5.0

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [X] Specification PDF in the form of wombat-spec-x.y.pdf
- [X] Specification HTML in the form of wombat-spec-x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [X] For a Release Review, a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification.
The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jakarta.servlet-api/5.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-servlet-tck-5.0.0.zip
- [X] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/servlet-api/issues/339
- [X] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

### CODE REVIEW

**Comment by kwsutter** [comment]
Hi @arjantijms.  I'm assigned Spec Mentor for this PR.  Feel free to reach out to me via Slack or email or this PR, if you have any questions.  Thanks!

**Comment by kwsutter** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

File: servlet/5.0/_index.md
Comment (kwsutter) [inline servlet/5.0/_index.md:13]: This TCK url should be the "final" location for the TCK.  Along the lines of https://download.eclipse.org/jakartaee/servlet/5.0/jakarta-servlet-tck-5.0.0.zip.  Also, don't forget to include the sig and sha links.

File: servlet/5.0/_index.md
Comment (kwsutter) [inline servlet/5.0/_index.md:15]: This should be the "final" maven coordinates for the API.  Along the lines of https://search.maven.org/artifact/jakarta.servlet/jakarta.servlet-api/5.0.0/jar

File: servlet/5.0/_index.md
Comment (arjantijms) [reply servlet/5.0/_index.md:13]: Yes, I know, although there's nothing there yet. Only https://download.eclipse.org/jakartaee/servlet/4.0/ 

Who is responsible for putting the TCK under /5.0?

File: servlet/5.0/_index.md
Comment (arjantijms) [reply servlet/5.0/_index.md:15]: Yes, I know this too ;) Still waiting for any +1 from the Servlet committers. I hope anyone will respond today.

Comment (arjantijms) [reply servlet/5.0/_index.md]: I've now put in the coordinates of the future link. It's of course not there yet.

File: servlet/5.0/_index.md
Comment (kwsutter) [reply servlet/5.0/_index.md]: That's part of the final approval process after the ballot is complete.  A member of the Spec Committee promotes the TCK to the proper location.  The requirement at this point is to put the expected location in the _index.md file and to provide the staging version of the TCK in the staging repo (which you already did).

File: servlet/5.0/_index.md
Comment (kwsutter) [reply servlet/5.0/_index.md]: Thanks.  Again, I would leave this in a staging repo until the ballot is complete.  The promotion of the maven central artifacts is left up to the project team.

File: servlet/5.0/_index.md
Comment (kwsutter) [reply servlet/5.0/_index.md]: This looks better now. Thanks.  But, we still need the sig and sha links as well...

**Comment by kwsutter** [comment]
I will start the review of the TCK material now.  I will also need a link to the Compatibility Certification request issue.  Thanks!

**Comment by kwsutter** [comment]
TCK Comments...
- The title is very generic (TCK User's Guide for Technology Implementors).  No mention of Servlet, except in the file name.
- The Copyright says Release 4.0 (instead of Release 5.0).
- The Copyright date says 2019.  Were there no updates for this release?  At least this page has to be updated due to the Servlet version increase.  But, maybe there were other updates as well?  Yeah, I see references to Servlet 4.0 throughout the Preface and other sections...

Thanks!

File: servlet/5.0/_index.md
Comment (arjantijms) [reply servlet/5.0/_index.md]: Thanks, done!

**Comment by arjantijms** [comment]
@kwsutter PR for TCK results here: https://github.com/eclipse-ee4j/glassfish/pull/23132

**Comment by kwsutter** [comment]
@arjantijms, on the Platform call today, we discussed the Automatic Module names.  We discovered a couple of components that have an "incorrect" automatic module name specified.  Servlet is one of them.  In the [servlet api pom file](https://github.com/eclipse-ee4j/servlet-api/blob/master/api/pom.xml), it is generating an automatic module name of "java.servlet".  To be consistent with the package name update, this should be "jakarta.servlet".  There is an active discussion whether these type of updates need to be done for Jakarta EE 9 or not.  For these cases where the `java` prefix is being used instead of `jakarta`, I think they should be.

This would require a respin of the Servlet 5.0.0 API jar file.

FYI, besides Servlet, we have also found Transactions, Authentication, and Authorization in the same boat.  There may be more.  @keilw is continuing the investigation.

**Comment by arjantijms** [comment]
@kwsutter 
> The title is very generic (TCK User's Guide for Technology Implementors). No mention of Servlet, except in the file name.

This actually seems to come from the general template. We can change this, but then it will change for all specs. In other words, not something for just Servlet to decide:

https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/Template/src/main/jbake/content/toc.adoc

> The Copyright date says 2019.

I think this may come from the general template as well.

**Comment by arjantijms** [comment]
I updated the local files anyway, but I'm not sure if this is correct. See PR: https://github.com/eclipse-ee4j/jakartaee-tck/pull/389

**Comment by kwsutter** [comment]
> I updated the local files anyway, but I'm not sure if this is correct. See PR: [eclipse-ee4j/jakartaee-tck#389](https://github.com/eclipse-ee4j/jakartaee-tck/pull/389)

Arjan, it looks like these common files were updated by Ed earlier this month:
https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/Template/src/main/jbake/content/title.adoc
https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/Template/src/main/jbake/content/title.inc

These type of updates should help with generating TCK User Guides with a title more specific to your Project.

**Comment by edbratt** [comment]
The updates @kwsutter refers to were made a year ago. If it would be helpful, I could submit a PR to revise these again for this CY and Jakarta EE 9 target release date Sept. 2020. Shall I?

**Comment by kwsutter** [comment]
> The updates @kwsutter refers to were made a year ago. If it would be helpful, I could submit a PR to revise these again for this CY and Jakarta EE 9 target release date Sept. 2020. Shall I?

Funny!  I saw July 11 and didn't even look at the year...  :-)  I thought these changes were in response to this https://github.com/eclipse-ee4j/jakartaee-tck/issues/364 created by Ivar.

**Comment by kwsutter** [comment]
@arjantijms None of the SHA values match.  I ran the SHA against the staged TCK and it was different from the one mentioned in the Certification Request and that one was different from the one in the TCK Results...  Maybe this all relates to your questions about promoting of the TCK zip files...  In any case, this needs to be resolved.

**Comment by arjantijms** [comment]
@kwsutter 

Yes, the TCK was re-staged and I indeed noticed a SHA change. I think the assumption here is to put all things in place, then as soon as the TCK is promoted, re-run it and update the SHAs.

I always calculate the SHA during the run of the TCK, so it's always from the actual one that is used for the run:

```
[javatest.batch] Jul 28, 2020 4:40:18 PM Finished executing all tests, wait for cleanup...
[javatest.batch] Jul 28, 2020 4:40:18 PM Harness done with cleanup from test run.
[javatest.batch] Total time = 3,241s
[javatest.batch] Setup time = 0s
[javatest.batch] Cleanup time = 0s
[javatest.batch] Test results: passed: 1,691
[javatest.batch] Results written to /home/jenkins/agent/workspace/2_servlet-run-tck-against-staged-build/servlettckwork/servlettck.
[javatest.batch] Report written to /home/jenkins/agent/workspace/2_servlet-run-tck-against-staged-build/servlettckreport/servlettck

BUILD SUCCESSFUL
Total time: 54 minutes 6 seconds
SHA256_GF=ab42efbbf6fa45f1db9f69cb3afa98f6f05b71113d265fc83034c31bc23d2318
SHA256_API=cbfbb73a69f9a4dbd76e69605824d60c7ef8576d3468059865e3c3b902aa200f
SHA256_TCK=c3d2e6200b16ab2b253022a5c11030eb5fd7b6ac6d244a49c83c9ddf73fa329b
Product_download=https://github.com/eclipse-ee4j/glassfish/releases/download/6.0.0-M2-servlet5/glassfish.zip
API_download=https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jakarta.servlet-api/5.0.0/jakarta.servlet-api-5.0.0.jar
TCK_download=https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip
OS4=3.12.0 NAME="Alpine Linux" ID=alpine VERSION_ID=3.12.0 PRETTY_NAME="Alpine Linux v3.12" HOME_URL="https://alpinelinux.org/" BUG_REPORT_URL="https://bugs.alpinelinux.org/"
JDK_VERSION=Picked up JAVA_TOOL_OPTIONS: -XX:+IgnoreUnrecognizedVMOptions -XX:+UnlockExperimentalVMOptions java version "1.8.0_202" Java(TM) SE Runtime Environment (build 1.8.0_202-b08) Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)
```

See here: https://ci.eclipse.org/servlet/job/2_servlet-run-tck-against-staged-build/lastSuccessfulBuild/artifact/servlet-tck/bin/summary.txt 

That TCK run was at 28-07, but then the TCK was re-staged at 30-07:

```
***********************************************************************************
***                        TCK bundle information                               ***
*** Name:       jakarta-servlet-tck-5.0.0.zip                                     ***
*** Bundle Copied to URL:    http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip ***
*** Date and size: date: 2020-07-30 17:40:06.000000000 +0000, size(b): 15127227        ***
*** SHA256SUM: 16cd8b951284b0871685ae7a6044e11cccf307c3220286e51da2bde1a6157500 ***
***                                                                             ***
***********************************************************************************
```

See https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0-tckinfo.txt

I can just re-run the TCK again now and when it finished copy the latest SHA everywhere, but of course if someone re-stages again they'll all be invalid again.

**Comment by arjantijms** [comment]
SHA updated here: https://github.com/eclipse-ee4j/glassfish/pull/23158 and in the CR, but I fear it's only a matter of time before they change again.

**Comment by kwsutter** [comment]
> SHA updated here: [eclipse-ee4j/glassfish#23158](https://github.com/eclipse-ee4j/glassfish/pull/23158) and in the CR, but I fear it's only a matter of time before they change again.

Thanks, @arjantijms.  I'm checking with @scottmarlow to see what can be done to help out with this immediate situation.  We need some type of staging solution that will work now (and maybe a better long term solution).

**Comment by arjantijms** [comment]
Ok, thx! Meanwhile it's not a super big problem to re-run the TCK and update the SHAs every few days. The biggest thing is probably that we need to be aware that this is happening as long as the TCK is not in the promoted folder.

**Comment by scottmarlow** [comment]
```
I'm checking with @scottmarlow to see what can be done to help out with this immediate situation. We need some type of staging solution that will work now (and maybe a better long term solution).
```

IMO, David's idea to include the [SHA in the staged TCK name will help with this](https://www.eclipse.org/lists/jakartaee-tck-dev/msg00858.html).

If an immediate change could help, we could update https://ci.eclipse.org/jakartaee-tck/job/eftl-standalonetck-build-run-900 to start including a timestamp in the TCK file names, which addresses the immediate concern.  We can then switch to David's script when that is available later (today or some later date).

As mentioned in the linked email thread, we could do something like:
```
mv $NAME ${NAME%.*}_$(date +%F-%H:%M).zip
```

Doing ^ would give us something like:
```
export NAME=jakarta-annotations-tck-2.0.0.zip
ls *.zip
jakarta-annotations-tck-2.0.0_2020-08-05-21:56.zip
```

If we make the ^ change today, then we know that the next time that we build/stage a TCK it will have the timestamp in the name which is very unlikely to be overwritten (unless the system clock is set back an hour and the TCK build occurs again.)  We could also hack the SHA into the filename as well, that would be even safer (still we could switch later to David's script when available).

We will make this change today if desire, just need to hear a +1 that this is the immediate fix.

CC: @alwin-joseph @gurunrao @RohitKumarJain @edbratt

**Comment by kwsutter** [comment]
@arjantijms Trying to clarify your specific situation.  Are your staged TCK files getting overwritten due to continued changes to the TCK on your part?  Or, they getting overwritten due to an external event that is not under your control?  If it's the former and you are still making updates to the TCK, then maybe you should wait to include a staged TCK and SHA value in the PR until you are ready.  If it's the latter, then let's get that into the discussion.

BTW, we have two discussions relating to this TCK staging and promoting process on the [public Spec Committee mailing list](https://accounts.eclipse.org/mailing-list/jakarta.ee-spec) and the [Platform TCK mailing list](https://accounts.eclipse.org/mailing-list/jakartaee-tck-dev).

**Comment by edbratt** [comment]
We're trying to work out a process by which the TCKs are completed and 
ready for ballot. This involves coordination with the TCK team and the 
API committer team -- the latter having final approval responsibility.

For Servlet, I think we're just about done. Once the refreshed zip file 
is created, the committer team (Arjan, in this case) needs to review the 
TCK contents, then indicate approval and notify the TCK team to promote 
it from the staged folder to the promoted folder.

Any SHA referenced before the TCK is moved to the promoted download 
folder is certain to change -- in part because the TCK bundle isn't 
confirmed to be done yet.

After that promotion into 'promoted' folder, the SHA will be stable and 
any subsequent builds would only be by specific request.

**Comment by alwin-joseph** [comment]
@kwsutter  Hi Kevin, 
I dont know if it is too late to ask this. For JakartaEE8 spec PRs it seems the https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/ directory was used as the staging folder for the standalone TCKs.
eg:  https://github.com/jakartaee/specifications/pull/79. 

We followed similar process for jakartaee9 but looks like the bundles in https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900 are being used as the staged TCKs in the spec PRs as per the template. The SHA sum for the same is being tried to match as well. Do you think it is ok to update the spec PR template now to start using bundles from https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted as the staged bundles for spec PRs. I think this will actually help avoid mismatch of  SHA sum too.

**Comment by kwsutter** [comment]
> @kwsutter Hi Kevin,
> I dont know if it is too late to ask this. For JakartaEE8 spec PRs it seems the https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/ directory was used as the staging folder for the standalone TCKs.
> eg: #79 

The use of the `promoted` directory is consistent with what @scottmarlow sent out to the team.  Until we get a better automated mechanism, we're going to stick with the "manual" operation.  So, once a team is happy with their TCK, they should contact the TCK team to promote their staged TCK to the `promoted` directory.  This is what should be specified in the checklist (although other variations may be used in a few exceptions like CDI, BV, Batch, etc).

**Comment by kwsutter** [comment]
Due to [@dblevins's PR](https://github.com/eclipse-ee4j/servlet-api/pull/350) getting merged for the javadoc, will you be updating the javadoc in this PR?

Also, will you be promoting the TCK to the "promoted" directory?  The SHA values are still out of sync.  Otherwise, this one is very close to starting a ballot... Thanks!

File: servlet/5.0/_index.md
Comment (jeanouii) [inline servlet/5.0/_index.md:10]: This should have jakarta- prefix

Please see PR https://github.com/eclipse-ee4j/servlet-api/pull/353

**Review by jeanouii — CHANGES_REQUESTED** [review]

**Comment by arjantijms** [comment]
> Due to @dblevins's PR getting merged for the javadoc, will you be updating the javadoc in this PR?

Yes, especially with the clarification (as discussed with EMO) in NOTICE.MD a re-stage and an update for the java doc PR is needed. I'll do this momentarily.

**Comment by arjantijms** [comment]
Just an update: outstanding NOTICE.MD changes have been merged as well as another last minute discovered error. Now going ahead to re-stage and re-run the TCK.

**Comment by alwin-joseph** [comment]
> Just an update: outstanding NOTICE.MD changes have been merged as well as another last minute discovered error. Now going ahead to re-stage and re-run the TCK.

I have triggered the job to refresh https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip . Will inform once it is done.

**Comment by arjantijms** [comment]
@alwin-joseph Thanks! 

API now restaged: https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jakarta.servlet-api/5.0.0/

File: servlet/5.0/_index.md
Comment (arjantijms) [reply servlet/5.0/_index.md]: Jakarta prefix added. Can you check again?

**Comment by alwin-joseph** [comment]
> > Just an update: outstanding NOTICE.MD changes have been merged as well as another last minute discovered error. Now going ahead to re-stage and re-run the TCK.
> 
> I have triggered the job (https://ci.eclipse.org/jakartaee-tck/job/eftl-standalonetck-build-run-900/62/) to refresh https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip . Will inform once it is done.

servlet tck is now refreshed at https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip

**Comment by arjantijms** [comment]
> > > Just an update: outstanding NOTICE.MD changes have been merged as well as another last minute discovered error. Now going ahead to re-stage and re-run the TCK.
> > 
> > 
> > I have triggered the job (https://ci.eclipse.org/jakartaee-tck/job/eftl-standalonetck-build-run-900/62/) to refresh https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip . Will inform once it is done.
> 
> servlet tck is now refreshed at https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-servlet-tck-5.0.0.zip

Perfect! TCK now running at https://ci.eclipse.org/servlet/job/2_servlet-run-tck-against-staged-build/35/

**Comment by arjantijms** [comment]
TCK executed and updated PR for GF TCK results done:

https://github.com/eclipse-ee4j/glassfish/pull/23197

**Comment by arjantijms** [comment]
@dblevins @jeanouii Can you check again whether your request for changes have been addressed sufficiently?

**Comment by jeanouii** [comment]
Reviewed again today and did not find any issue.

**Comment by arjantijms** [comment]
@jeanouii 
> Reviewed again today and did not find any issue.

Good! Can you remove the "changes requested" then? It's still active now ;)

**Review by jeanouii — APPROVED** [review]

**Comment by arjantijms** [comment]
@dblevins there's still a changes requested from you. Can you take a look?

**Comment by kwsutter** [comment]
@arjantijms So, you are not going to "promote" the TCK to the staged promotion directory?  https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/  This is the preferred location.  Although we could still do the official promotion from the staged directory, there is a chance that the SHA value could change or the TCK could accidentally be deleted or replaced.

**Comment by kwsutter** [comment]
@arjantijms In the Certification Request, shouldn't the TCK name be prefixed by "jakarta" instead of "eclipse"?
https://download.eclipse.org/jakartaee/servlet/5.0/eclipse-servlet-tck-5.0.0.zip  This is not consistent with what you put in the _index.md file:  https://download.eclipse.org/jakartaee/servlet/5.0/jakarta-servlet-tck-5.0.0.zip

**Comment by arjantijms** [comment]
>  In the Certification Request and TCK Results page, shouldn't the TCK name be prefixed by "jakarta" instead of "eclipse"?

Might be a copy/paste mistake, I'll take look right away.

**Review by kwsutter — COMMENTED** [review]
I have provided two comments that I would like a response to before approving and moving forward with a ballot.  Everything else looks good.  Thanks!

**Comment by alwin-joseph** [comment]
> @arjantijms So, you are not going to "promote" the TCK to the staged promotion directory? https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/ This is the preferred location. Although we could still do the official promotion from the staged directory, there is a chance that the SHA value could change or the TCK could accidentally be deleted or replaced.

Please let us know if no more changes are required to the servlet tck and it can moved to  https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/ or notify us on https://github.com/eclipse-ee4j/jakartaee-tck/issues/396. Thanks.

**Comment by arjantijms** [comment]
@kwsutter 

PR to update TCK result page: https://github.com/eclipse-ee4j/glassfish/pull/23201

**Comment by arjantijms** [comment]
@alwin-joseph 

No more changes are needed. It can be promoted. I'll also add a comment to the issue.

**Comment by alwin-joseph** [comment]
> @alwin-joseph
> 
> No more changes are needed. It can be promoted. I'll also add a comment to the issue.

Done. servlet TCK is now available at https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-servlet-tck-5.0.0.zip

**Comment by arjantijms** [comment]
Standalone Servlet TCK updated to run from promoted location: https://ci.eclipse.org/servlet/job/2_servlet-run-tck-against-staged-build/36/

**Comment by kwsutter** [comment]
> Standalone Servlet TCK updated to run from promoted location: https://ci.eclipse.org/servlet/job/2_servlet-run-tck-against-staged-build/36/

Results look good!  Thanks.

**Review by kwsutter — APPROVED** [review]
This all looks good now.  I'm assuming the PR for the TCK Results will get merged.  Everything else looks solid. Thanks, @arjantijms!

**Comment by kwsutter** [comment]
From what I can tell, @dblevins suggestions (PR) have been incorporated.  But, I will wait for confirmation before initiating a ballot.  Everything looks good from my perspective as mentor.

**Comment by arjantijms** [comment]
@dblevins can you take a look again?

**Comment by kwsutter** [comment]
> @dblevins can you take a look again?

I have pinged @dblevins via Slack.  If I don't hear back today, I will go ahead with initiating the ballot (as the official Mentor for this Specification).  As I mentioned, it looks like David's previous comments and PR have been addressed.  But, I'd like to give him one last chance.  Thanks!

**Review by dblevins — APPROVED** [review]
Sorry for the delay.  Looks great!  Thank you for incorporating the PR! 👍

**Comment by arjantijms** [comment]
Release Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=566965
CQ/IP log Bug: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22588

**Comment by arjantijms** [comment]
Ballot: https://eclipse.org/lists/jakarta.ee-spec/msg00898.html

**Comment by kwsutter** [comment]
* On ballot completion, the specification committee mentor:
 - [x] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>


 ----
 - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
 - [ ] approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] merge any final release branch as appropriate for the branch management for the project.

**Pull request merged**
